### PR TITLE
Adding support for the values comming from the OM2M devices

### PR DESCRIPTION
**Pull request by dkurzaj**
Adds the support of the IoT devices represented in OM2M. The facts deducted are only valuable for the specified devices and conditions.

### CODE REVIEW

File: pdg/src/OM2MObjectReader.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    //put the same orientation and positions as the previous time
+    curObject->setOrientation(objectOrientation);
+    curObject->setPosition(objectPosition);
+
+    // set the new value of the OM2M object
+    curObject->setValue(msg->iot_data.data.value);
+
+    // set the time
+    unsigned long milli_sec = msg->iot_data.header.stamp.sec * 1000;
+    curObject->setTime(milli_sec);
```
Comment (Greg8978) [inline pdg/src/OM2MObjectReader.cpp]: I think usually we have time in microseconds in TOASTER.

File: pdg/src/main.cpp [hunk 2]
```diff
@@ -719,27 +743,475 @@ int main(int argc, char** argv) {
...
+                                fact_msg.factObservability = 0.5;
+                                fact_msg.time = it->second->getTime();
+                                fact_msg.valueType = 0;
+                                fact_msg.stringValue = value_value.c_str();
+
+                                factList_msg.factList.push_back(fact_msg);
+                            }
+                            else if (value_value == "false") {
+                                //Fact message
+                                fact_msg.property = "FanOFF";
```
Comment (Greg8978) [inline pdg/src/main.cpp]: The property should be the same: FanON, only the value should change.

**Review by Greg8978 — CHANGES_REQUESTED** [review]
The code is good and thank you for corrections outside of your contribution :)

This being said, I'm not very satisfied with the part checking which IOT object it is.

Couldn't we have xml file to associate the element to the fact generation?
There is an example here (https://github.com/Greg8978/hatm/blob/master/data/list_nodes.xml), where I associated task names to there speech, you can use the same idea to associate object name to a property...

Something like

for each IOTObject{
  property = loadPropertyFromXml(objectID))
  Value = value.c_str()
}

Also the property should not change, only the value.
e.g
property = FanOn | value = true (fan is on) 
property = FanOn | value = false (fan is off)
property = LightColor  | value = yellow (color is yellow)

...

Let me know if you need to discuss further about that :)

**Comment by Greg8978** [comment]
I forgot but the advantage of the xml here is also to add IOT object without need of recompiling pdg :)

**Review by Greg8978 — APPROVED** [review]

**Comment by Greg8978** [comment]
If you can update the wiki (maybe create IOT page with a link in pdg page?)
to explain how to use pdg with IOT objects and how to edit the xml to add IOTobjects/facts it would be perfect :)

Cheers!

**Comment by dkurzaj** [comment]
Okay sure !

**Pull request merged**